tic_tac_toe/game.py

```diff
@@ -1,30 +1,56 @@
 from tic_tac_toe.board import Board
+from tic_tac_toe.player import Player, Human
 from tic_tac_toe.views import CommandLineBoardPresenter
+from tic_tac_toe.user_messages import UserMessages
 from tic_tac_toe.printer import Printer
 from tic_tac_toe.user_input import UserInput
 from tic_tac_toe.errors import (
     Errors,
     InvalidBoardIndexError,
     InputNotNumericError,
-    PositionAlreadyTaken,
+    PositionAlreadyTakenError,
 )
 
 
 def run():
     board = Board()
+    player = Human(token=None)
     view = CommandLineBoardPresenter()
     printer = Printer()
+    user_messages = UserMessages(printer)
     user_selection = UserInput()
     errors = Errors(printer)
 
-    while True:
+    user_messages.countdown()
+
+    user_messages.logo()
+
+    user_messages.instructions_option()
+
+    if user_selection.process_input() == 0:
```

Comment:
The zero here stands out as a "magic number"—a value that, through context, can be understood to have some important meaning, but it's not immediately obvious _why_. "Why zero?" Essentially you're bubbling up an implementation detail higher than it probably ought to go. Contrast this with below, where you have a really nice high-level description of what's going on where you write `while not board.game_over()`—that's a much nicer way of expressing intent. If someone wants to figure out how to determine whether or not the game is over, they can dive into that class/method, but if you don't care about the details and are just staying at this high level, it is more expressive.

Reply:
That makes a lot of sense. Originally when I was putting together the prompt that this if statement is a part of, ("would you like to view the instructions?"), I wanted the user to type 'yes' or 'no', but that would have broken the process_input() function that only lets the user enter a number. Rather than rewrite that, I just decided to have the user enter 1 for yes and 2 for no, but the method substracts 1 from whatever the input is so that it corresponds with the board list indices.

So, yeah, if I'm understanding what you're saying, you're definitely right. The process_input() function is handling input at a higher level than it's meant to.